frontend_tests/node_tests/voting_widget.js

```diff
@@ -76,4 +84,291 @@ run_test('poll_data_holder my question', () => {
+    data = data_holder.get_widget_data();
+
+    const comment_outbound_event = data_holder.handle.new_comment.outbound('new comment');
+    assert.deepEqual(comment_outbound_event, {
+        type: 'new_comment',
+        idx: 2,
+        comment: 'new comment',
+    });
+
+    const new_question = 'Any new plan?';
+    const question_outbound_event = data_holder.handle.question.outbound(new_question);
+    assert.deepEqual(question_outbound_event, {
+        type: 'question',
+        question: new_question,
+    });
+
+    const vote_outbound_event = data_holder.handle.vote.outbound('99,1');
+    assert.deepEqual(vote_outbound_event, { type: 'vote', key: '99,1', vote: -1 });
+
+    vote_event = {
+        type: 'vote',
+        key: '99,1',
+        vote: -1,
+    };
+
+    data_holder.handle_event(sender_id, vote_event);
+    data = data_holder.get_widget_data();
+
+    assert.deepEqual(data, {
+        comments: [
+            {
+                comment: 'release now',
+                names: '',
+                count: 0,
+                key: '99,1',
+            },
+        ],
+        question: 'best plan?',
+    });
+});
+
+run_test('activate another person poll', () => {
+    people.is_my_user_id = return_false;
+    templates.render = (template_name) => {
+        if (template_name === 'poll-widget') {
+            return 'poll-widget';
+        } else if (template_name === 'poll-widget-results') {
+            return 'poll-widget-results';
+        }
+    };
+
+    const widget_elem = $('<div>').addClass('widget-content');
+    let out_data;
+    const callback = (data) => {
+        out_data = data;
+    };
+    const opts = {
+        elem: widget_elem,
+        callback: callback,
+        message: {
+            sender_id: 100,
+        },
+    };
+
+    const poll_comment = $.create('button.poll-comment');
+    const poll_comment_input = $.create('input.poll-comment');
+    const widget_comment_container = $.create('ul.poll-widget');
+
+    const poll_question = $.create('button.poll-question');
+    const poll_question_input = $.create('input.poll-question');
+    const poll_question_header = $.create('.poll-question-header');
+    const poll_question_container = $.create('.poll-question-bar');
+
+    const poll_vote_button = $.create('button.poll-vote');
+
+    widget_elem.set_find_results('button.poll-comment', poll_comment);
+    widget_elem.set_find_results('input.poll-comment', poll_comment_input);
+    widget_elem.set_find_results('button.poll-question', poll_question);
+    widget_elem.set_find_results('input.poll-question', poll_question_input);
+    widget_elem.set_find_results('.poll-question-header', poll_question_header);
+    widget_elem.set_find_results('ul.poll-widget', widget_comment_container);
+    widget_elem.set_find_results('button.poll-vote', poll_vote_button);
+    widget_elem.set_find_results('.poll-question-bar', poll_question_container);
+
+    poll_comment.on = (event, func) => {
```

Comment:
The way this is structured is a little bit confusing, although I think I get what you're doing.  Can you either add a comment to the effect of using-stub-click-handler-to-test-both-conditions or maybe think of a more clear way to achieve this?  (I've done similar things in the past, but usually there's a way to tidy this up a bit.)